modules/gapi/test/internal/gapi_int_dynamic_graph.cpp

```diff
@@ -0,0 +1,229 @@
+
+        cv::GRunArgs in_vector;
+        in_vector += cv::gin(in_mat1);
+        in_vector += cv::gin(in_mat2);
+
+        cv::Mat out_mat1, out_mat2;
+        cv::GRunArgsP out_vector;
+        out_vector += cv::gout(out_mat1);
+        out_vector += cv::gout(out_mat2);
+
+        cc.apply(std::move(in_vector), std::move(out_vector));
+
+        // OCV ref code
+        cv::Mat cv_out_mat1, cv_out_mat2;
+        cv::resize(in_mat1, cv_out_mat1, szOut);
+        cv::resize(in_mat2, cv_out_mat2, szOut);
+
+        EXPECT_EQ(0, cvtest::norm(out_mat1, cv_out_mat1, NORM_INF));
+        EXPECT_EQ(0, cvtest::norm(out_mat2, cv_out_mat2, NORM_INF));
+    }
+
+    TEST(DynamicGraph, Streaming)
+    {
+        cv::GComputation cc([&](){
+            cv::Size szOut(4, 4);
+
+            cv::GMat in1;
+            cv::GProtoInputArgs ins = GIn(in1);
+
+            cv::GMat in2;
+            ins += GIn(in2);
+
+            cv::GMat out1 = cv::gapi::resize(in1, szOut);
+            cv::GProtoOutputArgs outs = GOut(out1);
+
+            cv::GMat out2 = cv::gapi::resize(in2, szOut);
+            outs += GOut(out2);
+
+            return cv::GComputation(std::move(ins), std::move(outs));
+        });
+
+        EXPECT_NO_THROW(cc.compileStreaming(cv::compile_args(cv::gapi::core::cpu::kernels())));
+    }
+
+    TEST(DynamicGraph, StreamingAccuracy)
+    {
+        cv::Size szOut(4, 4);
+        cv::GComputation cc([&](){
+            cv::GMat in1;
+            cv::GProtoInputArgs ins = GIn(in1);
+
+            cv::GMat in2;
+            ins += GIn(in2);
+
+            cv::GMat out1 = cv::gapi::resize(in1, szOut);
+            cv::GProtoOutputArgs outs = GOut(out1);
+
+            cv::GMat out2 = cv::gapi::resize(in2, szOut);
+            outs += GOut(out2);
+
+            return cv::GComputation(std::move(ins), std::move(outs));
+        });
+
+        // G-API test code
+        cv::Mat in_mat1( 8,  8, CV_8UC3);
+        cv::Mat in_mat2(16, 16, CV_8UC3);
+        cv::randu(in_mat1, cv::Scalar::all(0), cv::Scalar::all(255));
+        cv::randu(in_mat2, cv::Scalar::all(0), cv::Scalar::all(255));
+
+        cv::GRunArgs in_vector;
+        in_vector += cv::gin(in_mat1);
+        in_vector += cv::gin(in_mat2);
+
+        cv::Mat out_mat1, out_mat2;
+        cv::GRunArgsP out_vector;
+        out_vector += cv::gout(out_mat1);
+        out_vector += cv::gout(out_mat2);
+
+        auto stream = cc.compileStreaming(cv::compile_args(cv::gapi::core::cpu::kernels()));
+        stream.setSource(std::move(in_vector));
+
+        stream.start();
+        stream.pull(std::move(out_vector));
+        stream.stop();
+
+        // OCV ref code
+        cv::Mat cv_out_mat1, cv_out_mat2;
+        cv::resize(in_mat1, cv_out_mat1, szOut);
+        cv::resize(in_mat2, cv_out_mat2, szOut);
+
+        EXPECT_EQ(0, cvtest::norm(out_mat1, cv_out_mat1, NORM_INF));
+        EXPECT_EQ(0, cvtest::norm(out_mat2, cv_out_mat2, NORM_INF));
+    }
+} // namespace opencv_test
```

Comment:
Is it right I see only `cv::GMat` tested here? What about other types?

Reply:
Add some typed tests for `cv::GMatP` ,  `cv::GScalar` etc.